test/controllers/locationOfGoods/IdentificationControllerSpec.scala

```diff
@@ -19,100 +19,96 @@ package controllers.locationOfGoods
-  private val mode                     = NormalMode
-  private lazy val identificationRoute = routes.IdentificationController.onPageLoad(lrn, mode).url
+  private val id1                                                                     = Arbitrary.arbitrary[LocationOfGoodsIdentification].sample.value
+  private val id2                                                                     = Arbitrary.arbitrary[LocationOfGoodsIdentification].sample.value
+  private val ids                                                                     = Seq(id1, id2)
+  private val formProvider                                                            = new EnumerableFormProvider()
+  private val form                                                                    = formProvider[LocationOfGoodsIdentification]("locationOfGoods.identification", ids)
+  private val mode                                                                    = NormalMode
+  private lazy val identificationRoute                                                = routes.IdentificationController.onPageLoad(lrn, mode).url
+  private val mockLocationIdentifierService: LocationOfGoodsIdentificationTypeService = mock[LocationOfGoodsIdentificationTypeService]
 
   override def guiceApplicationBuilder(): GuiceApplicationBuilder =
     super
       .guiceApplicationBuilder()
       .overrides(bind(classOf[LocationOfGoodsNavigatorProvider]).toInstance(fakeLocationOfGoodsNavigatorProvider))
+      .overrides(bind(classOf[LocationOfGoodsIdentificationTypeService]).toInstance(mockLocationIdentifierService))
 
-  "LocationOfGoodsIdentification Controller" - {
-
-    "when value is inferred" - {
-      "must redirect to next page" in {
-        val userAnswers = emptyUserAnswers.setValue(LocationTypePage, AuthorisedPlace)
-        setExistingUserAnswers(userAnswers)
-
-        val request = FakeRequest(GET, identificationRoute)
-
-        val result = route(app, request).value
-
-        status(result) mustEqual SEE_OTHER
-
-        redirectLocation(result).value mustEqual onwardRoute.url
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    reset(mockLocationIdentifierService)
+    when(mockLocationIdentifierService.getLocationOfGoodsIdentificationTypes(any())(any())).thenReturn(Future.successful(ids))
+  }
 
-        val userAnswersCaptor: ArgumentCaptor[UserAnswers] = ArgumentCaptor.forClass(classOf[UserAnswers])
-        verify(mockSessionRepository).set(userAnswersCaptor.capture())(any())
-        userAnswersCaptor.getValue.getValue(InferredIdentificationPage) mustBe AuthorisationNumber
-        userAnswersCaptor.getValue.get(IdentificationPage) must not be defined
-      }
-    }
+  "LocationOfGoodsIdentification Controller" - {
 
     "must return OK and the correct view for a GET" in {
+      forAll(arbitrary[LocationType]) {
+        locationType =>
+          val ua = emptyUserAnswers.setValue(LocationTypePage, locationType)
```

Comment:
Looks like we no longer need to set a value in LocationTypePage. Other than mocking a response for `mockLocationIdentifierService.getLocationOfGoodsIdentificationTypes` this class can pretty much go back to how it was